hvp.php

```diff
@@ -561,26 +561,43 @@ public function alterLibrarySemantics(&$semantics, $name, $majorVersion, $minorV
   public function loadContent($id) {
     global $DB;
 
-    // TODO: Verify that this work with other databases than MySQL. I suspect
-    // that camelCase won't work for e.g. PostgreSQL
-    $content = $DB->get_record_sql(
+    $data = $DB->get_record_sql(
         "SELECT hc.id
-              , hc.json_content AS params
+              , hc.name
+              , hc.json_content
               , hc.filtered
               , hc.slug
-              , hc.embed_type AS embedType
+              , hc.embed_type
               , hc.disable
-              , hl.id AS libraryId
-              , hl.machine_name AS libraryName
-              , hl.major_version AS libraryMajorVersion
-              , hl.minor_version AS libraryMinorVersion
-              , hl.embed_types AS libraryEmbedTypes
-              , hl.fullscreen AS libraryFullscreen
+              , hl.id AS library_id
+              , hl.machine_name
+              , hl.major_version
+              , hl.minor_version
+              , hl.embed_types
+              , hl.fullscreen
         FROM {hvp} hc
         JOIN {hvp_libraries} hl ON hl.id = hc.main_library_id
         WHERE hc.id = ?", array($id)
     );
 
+    // Some databases do not support camelCase, so we need to manually
+    // map the values to the camelCase names used by the H5P core.
+    $content = array(
+        'id' => $data->id,
+        'title' => $data->name,
+        'params' => $data->json_content,
+        'filtered' => $data->filtered,
+        'slug' => $data->slug,
+        'embedType' => $data->embed_type,
+        'disable' => $data->disable,
+        'libraryId' => $data->library_id,
+        'libraryName' => $data->machine_name,
+        'libraryMajorVersion' => $data->major_version,
+        'libraryMinorVersion' => $data->minor_version,
+        'libraryEmbedTypes' => $data->embed_types,
+        'libraryFullscreen' => $data->fullscreen,
```

Comment:
Yes, this is far from ideal.
I suggest we get it working before we start any refactoring.